builtin/rebase.c

```diff
@@ -817,6 +823,19 @@ int cmd_rebase(int argc, const char **argv, const char *prefix)
 		}
 	}
 
+	if (gpg_sign) {
+		struct strbuf buf2 = STRBUF_INIT;
+
+		strbuf_reset(&buf);
+		strbuf_reset(&buf2);
+		strbuf_addf(&buf2, "--%s", gpg_sign);
```

Comment:
Why prefix it with `--`? The shell code does not do that:

```sh
	--gpg-sign=*)
		gpg_sign_opt="-S${1#--gpg-sign=}"
```

It strips off the `--gpg-sign=` prefix of the command-line option, and then prefixes the remaineder with `-S`.